### DISCUSSION TITLE
AttributeError: module 'tensorly.backend' has no attribute 'decomposition'

### DISCUSSION THREAD

**Discussion by leohermet1** [tensorly/tensorly#413] [Q&A]
Hi, 
I'm using tensorly and I had this problem for a while now.

On linux, mac or windows,
New environment with python version 3.6.8 or 3.9.12 or 3.10.2 ,
i run pip3 _install -U tensorly_ or _pip3 install tensorly_  
`import tensorly`
and when I want to use the decomposition module with tensorly.decomposition() it says 
`AttributeError: module 'tensorly.backend' has no attribute 'decomposition'`
Even tho when I print the list of all the modules of tensorly in the same script there is 'decomposition' in it.

But when I'm using `from tensorly.decomposition import parafac2` it's working.

**Comment by JeanKossaifi** [comment]
I think the issue is that `tensorly.decomposition` is not imported by default, can you try to just add `import tensorly.decomposition` first?

**Reply by leohermet1** [reply]
Yes it's working by just adding `import tensorly.decomposition` .
I just wanted to report it since I got stuck with this for a long time not knowing what was wrong.
